packages/webapp/components/layouts/AccountLayout/Profile/VerifyWorkEmail.tsx

```diff
@@ -0,0 +1,128 @@
+import classNames from 'classnames';
+import { MailIcon, VIcon } from '@dailydotdev/shared/src/components/icons';
+import { TextField } from '@dailydotdev/shared/src/components/fields/TextField';
+import {
+  Button,
+  ButtonVariant,
+} from '@dailydotdev/shared/src/components/buttons/Button';
+import React, { ReactElement, useContext, useState } from 'react';
+import useTimer from '@dailydotdev/shared/src/hooks/useTimer';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { gqlClient } from '@dailydotdev/shared/src/graphql/common';
+import {
+  ADD_USER_COMPANY_MUTATION,
+  VERIFY_USER_COMPANY_CODE_MUTATION,
+} from '@dailydotdev/shared/src/graphql/users';
+import { UserCompany } from '@dailydotdev/shared/src/lib/userCompany';
+import {
+  generateQueryKey,
+  RequestKey,
+} from '@dailydotdev/shared/src/lib/query';
+import AuthContext from '@dailydotdev/shared/src/contexts/AuthContext';
+import { labels } from '@dailydotdev/shared/src/lib';
+import { useToastNotification } from '@dailydotdev/shared/src/hooks';
+import { AccountSecurityDisplay as Display, CommonTextField } from '../common';
+import { AccountPageContainer } from '../AccountPageContainer';
+
+const VerifyWorkEmail = ({
+  onSwitchDisplay,
+  workEmail,
+}: {
+  onSwitchDisplay: (display: Display) => void;
+  workEmail?: string;
+}): ReactElement => {
+  const queryClient = useQueryClient();
+  const { user } = useContext(AuthContext);
+  const { displayToast } = useToastNotification();
+  const [hint, setHint] = useState<string>();
+  const [code, setCode] = useState<string>();
+  const { timer, setTimer, runTimer } = useTimer(null, 0);
+
+  const { mutate: onSubmitCode } = useMutation(
+    () => gqlClient.request(ADD_USER_COMPANY_MUTATION, { email: workEmail }),
+    {
+      onSuccess: () => {
+        setTimer(60);
+        runTimer();
+      },
+      onError: () => displayToast(labels.error.generic),
+    },
+  );
+
+  const { mutate: verifyUserCompanyCode } = useMutation(
+    ({ email, code: submittedCode }: { email: string; code: string }) =>
```

Comment:
`submittedCode` doesn't work, get GraphQL error
> "Variable "$code" of required type "String!" was not provided."
![image](https://github.com/user-attachments/assets/5e95f8c3-baa9-44ec-bdee-150c03518756)

Reply:
Oh my bad, scoped in during lint fixes